prototypes/ScheduledMerges.hs

```diff
@@ -93,58 +93,79 @@ type Levels s = [Level s]
 -- completed, the resulting run will become a resident run at this level.
 data Level s = Level !(IncomingRun s) ![Run]
 
--- | An additional optional last level, created as a result of 'union'. It can
--- not only contain an ongoing merge of multiple runs, but a nested tree of
--- merges. See Note [Table Unions].
-data UnionLevel s = NoUnion
-                    -- | We track the debt to make sure it never increases.
-                  | Union !(MergingTree s) !(STRef s Debt)
+-- | We represent single runs specially, rather than putting them in as a
+-- 'CompletedMerge'. This is for two reasons: to see statically that it's a
+-- single run without having to read the 'STRef', and secondly to make it easier
+-- to avoid supplying merge credits. It's not essential, but simplifies things
+-- somewhat.
+data IncomingRun s = Merging !MergePolicy !(MergingRun LevelMergeType s)
+                   | Single  !Run
 
 -- | The merge policy for a LSM level can be either tiering or levelling.
 -- In this design we use levelling for the last level, and tiering for
 -- all other levels. The first level always uses tiering however, even if
--- it's also the last level. So 'MergePolicy' and 'MergeLastLevel' are
+-- it's also the last level. So 'MergePolicy' and 'IncomingMergeType' are
```

Comment:
`IncomingMergeType` does not seem to exist

Reply:
Good catch. I renamed it to `LevelMergeType`, but seems like my search missed this occurrence because it considers the tick `'` as part of identifier names. I'll fix it as part of another PR.